admin/resources/js/components/H2Components/Button.tsx

```diff
@@ -0,0 +1,52 @@
+import React from "react";
+
+export interface ButtonProps {
+  children: React.ReactNode;
+  borderColor:
+    | "lightpurple"
+    | "lightnavy"
+    | "darkgold"
+    | "white"
+    | "[light]white[0]";
+  backgroundColor:
+    | "lightpurple"
+    | "[light]lightpurple[.1]"
+    | "darkpurple"
+    | "lightnavy"
+    | "[light]lightnavy[.1]"
+    | "darkgold"
+    | "[light]darkgold[.1]"
+    | "white"
+    | "[light]white[0]";
+  fontColor: "black" | "white" | "lightpurple" | "lightnavy" | "darkgold";
+  onClick?: () => void;
+}
+
+const Button: React.FC<ButtonProps> = ({
+  children,
+  borderColor,
+  backgroundColor,
+  fontColor,
+  onClick,
+}): React.ReactElement => {
+  return (
+    <button
+      type="button"
+      data-h2-border={`b(${borderColor}, all, solid, s)`}
+      data-h2-radius="b(s)"
+      data-h2-padding="b(top-bottom, xs) b(right-left, s)"
+      data-h2-bg-color={`b(${backgroundColor})`}
+      data-h2-font-family="b(sans)"
```

Comment:
I'm unsure about including font-family on these H2 components. Shouldn't we just let them inherit, and set the font at a higher level?

Reply:
It totally depends on if other fonts will be used on the project. If there's only ever one font used, then yes, inheritance from the html element is probably the easiest. Will that be the case?